docs/decisions/0002-lti-1p3-variables.rst

```diff
@@ -0,0 +1,161 @@
+LTI 1.3 variable definition
+---------------------------
+
+Status
+======
+
+Provisional
+
+Context
+=======
+
+The LTI 1.3 specification defines multiple variables related to course, context, problem, student id and PII.
+
+The goal of this ADR is to provide well defined mappings from variables available in the platform and
+variables required by LTI 1.3 launches.
+
+Decisions
+=========
+
+The following section go over required and optional variables in each supported claim in the LTI 1.3 implementation.
+This ADR should be updated when new claims are added.
+
+Deployment ID
+~~~~~~~~~~~~~
+
+.. list-table::
+   :widths: auto
+   :header-rows: 1
+
+   * - LTI 1.3 variable name
+     - Open edX
+     - Reasoning
+   * - https://purl.imsglobal.org/spec/lti/claim/deployment_id
+     - "1" (fixed value)
+     - We're using single-tenant deployments, were each key is used for a single XBlock. Even when moving up to the course level,
+       we'll can keep this setup since multi-tenancy will only be required for instance wide configuration.
+
+See http://www.imsglobal.org/spec/lti/v1p3/#single-tenant-tool-registered-and-deployed-once for more details.
+
+Roles
+~~~~~
+The roles claim takes in a list of LTI 1.3 compliant roles from http://www.imsglobal.org/spec/lti/v1p3/#role-vocabularies.
+The claim name is: https://purl.imsglobal.org/spec/lti/claim/roles.
+
+The mappings from edX platform are shown in the table below:
+
+.. list-table::
+   :widths: auto
+   :header-rows: 1
+
+   * - Open edX role
+     - LTI 1.3 Roles included
+     - Reasoning
+   * - guest
+     - Empty
+     - Guests are users not logged in, they shoudn't be able to access LTI content.
+   * - student
+     - 'http://purl.imsglobal.org/vocab/lis/v2/institution/person#Student'
+     - Students only have permission to view and interact with the LTI content.
+   * - instructor
+     - 'http://purl.imsglobal.org/vocab/lis/v2/institution/person#Instructor'
+       'http://purl.imsglobal.org/vocab/lis/v2/institution/person#Student'
+     - Instructor have both instructor and student access to a tool. No admin permissions.
+   * - staff
+     - 'http://purl.imsglobal.org/vocab/lis/v2/system/person#Administrator'
+       'http://purl.imsglobal.org/vocab/lis/v2/institution/person#Instructor'
+       'http://purl.imsglobal.org/vocab/lis/v2/institution/person#Student'
+     - A staff user should be able to modify the tool settings and have full access to the content and settings.
```

Comment:
We should say something here about why we are choosing `http://purl.imsglobal.org/vocab/lis/v2/system/person#Administrator` instead of `http://purl.imsglobal.org/vocab/lis/v2/institution/person#Administrator`

Reply:
@nedbat I've picked the most generic one to give the maximum amount of permissions to a course creator. We should probably reevaluate these roles once we have tested the platform with a few LTI tools. 

I've just noticed that the spec mentions:
```
The role vocabularies are derived from the LIS 2.0 specification [LIS-20]. LTI divides them into core and non-core roles. Core roles are those which are most likely to be relevant within LTI and hence vendors should support them by best practice. Vendors may also use the non-core rules, but they may not be widely used.
```
But the cores role are just `Administrator` and `None`: http://www.imsglobal.org/spec/lti/v1p3/#lis-vocabulary-for-system-roles